ui/src/utils/plugins.js

```diff
@@ -490,8 +499,7 @@ export const genericUtilPlugin = {
     }
 
     app.config.globalProperties.$toBase64AndURIEncoded = function (text) {
-      const base64regex = /^([0-9a-zA-Z+/]{4})*(([0-9a-zA-Z+/]{2}==)|([0-9a-zA-Z+/]{3}=))?$/
-      if (base64regex.test(text)) {
+      if (isBase64(text)) {
```

Comment:
@Pearl1594 
thanks for the update.
with your latest change, do we still need this ?
```
      if (isBase64(text)) {
        return text
      }
```

Reply:
Do we not want the text that's not encoded to be encoded and saved in the DB .. I thought the idea of the flag was the prevent encoding an already encoded string

Reply:
@Pearl1594 
I think the logic can be simple as below
- if checkbox is ticked, pass the origin input
- if checkbox is not ticked, (Base64AndURI) encode the input and pass to API

Reply:
oh, wait @Pearl1594 
the method `toBase64AndURIEncoded` is being used in some other files..

```
ui/src/utils/plugins.js:    app.config.globalProperties.$toBase64AndURIEncoded = function (text) {
ui/src/views/compute/CreateAutoScaleVmGroup.vue:          createVmGroupData.userdata = this.$toBase64AndURIEncoded(values.userdata)
ui/src/views/compute/DeployVM.vue:          deployVmData.userdata = this.$toBase64AndURIEncoded(values.userdata)
ui/src/views/compute/DeployVnfAppliance.vue:          createVnfAppData.userdata = this.$toBase64AndURIEncoded(values.userdata)
ui/src/views/compute/EditVM.vue:          params.userdata = this.$toBase64AndURIEncoded(values.userdata)
ui/src/views/compute/RegisterUserData.vue:        params.userdata = this.$toBase64AndURIEncoded(values.userdata)
ui/src/views/compute/ResetUserData.vue:        params.userdata = this.$toBase64AndURIEncoded(values.userdata)
```

Reply:
> @Pearl1594 I think the logic can be simple as below
> 
>     * if checkbox is ticked, pass the origin input
> 
>     * if checkbox is not ticked, (Base64AndURI) encode the input and pass to API

That is what is happening in RegisterUserdata component, that we see if it's unchecked and call the encode method.

Reply:
@Pearl1594 @weizhouapache is there a consensus on changes here?

Reply:
I had forgotten about this PR - however, @weizhouapache 's input would be appreciated at this point.